### ISSUE TITLE
Got CMake error while build on linux

### ISSUE THREAD

**Issue by meepeek** [OpenApoc/OpenApoc#318]
Please help, I followed the instruction to build for linux and got error. I am using Deepin, based on Debian.

I have got an error for not having CMakeLists.txt in dependencies/physfs so I copied one frome the root directory, then I ran the command ``` cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo .. ``` in build directory again and got the following error in build/CMakeFiles/CMakeError.log:

```
Determining if the pthread_create exist failed with the following output:
Change Dir: /home/thee/games/OpenApoc/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_4f564/fast"
/usr/bin/make -f CMakeFiles/cmTC_4f564.dir/build.make CMakeFiles/cmTC_4f564.dir/build
make[1]: Entering directory '/home/thee/games/OpenApoc/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_4f564.dir/CheckSymbolExists.c.o
/usr/bin/cc    -fvisibility=hidden    -o CMakeFiles/cmTC_4f564.dir/CheckSymbolExists.c.o   -c /home/thee/games/OpenApoc/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_4f564
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_4f564.dir/link.txt --verbose=1
/usr/bin/cc   -fvisibility=hidden     CMakeFiles/cmTC_4f564.dir/CheckSymbolExists.c.o  -o cmTC_4f564 -rdynamic 
CMakeFiles/cmTC_4f564.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x1b): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_4f564.dir/build.make:97: recipe for target 'cmTC_4f564' failed
make[1]: *** [cmTC_4f564] Error 1
make[1]: Leaving directory '/home/thee/games/OpenApoc/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_4f564/fast' failed
make: *** [cmTC_4f564/fast] Error 2

File /home/thee/games/OpenApoc/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

```
This is what it looks like at the beginning of my console (It too long so I didn't copy them all)
```
thee@joke-PC:~/games/OpenApoc/build$ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
Using CD at "/home/thee/games/OpenApoc/data/cd.iso"
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   locale
-- Using <experimental/filesystem>
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   locale
--   system
--   program_options
Searching for SDL2 in 
Found SDL2 at /usr/lib/x86_64-linux-gnu/libSDL2.so
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   program_options
Using CD at "/home/thee/games/OpenApoc/data/cd.iso"
CMake Error at dependencies/CMakeLists.txt:82 (add_subdirectory):
  add_subdirectory given source "library" which is not an existing directory.


CMake Error at dependencies/CMakeLists.txt:83 (add_subdirectory):
  add_subdirectory given source "framework" which is not an existing
  directory.


CMake Error at dependencies/CMakeLists.txt:84 (add_subdirectory):
  add_subdirectory given source "forms" which is not an existing directory.


CMake Error at dependencies/CMakeLists.txt:85 (add_subdirectory):
  add_subdirectory given source "game/state" which is not an existing
  directory.


CMake Error at dependencies/CMakeLists.txt:86 (add_subdirectory):
  add_subdirectory given source "game/ui" which is not an existing directory.


CMake Error at dependencies/CMakeLists.txt:87 (add_subdirectory):
  add_subdirectory given source "tools" which is not an existing directory.


CMake Error at dependencies/CMakeLists.txt:88 (add_subdirectory):
  add_subdirectory given source "dependencies" which is not an existing
  directory.


CMake Error at dependencies/CMakeLists.txt:119 (add_subdirectory):
  add_subdirectory given source "tests" which is not an existing directory.


Found xmllint /usr/bin/xmllint
CMake Error: File /home/thee/games/OpenApoc/dependencies/version.h.in does not exist.
CMake Error at dependencies/CMakeLists.txt:238 (configure_file):
  configure_file Problem configuring file


CMake Error at CMakeLists.txt:93 (ADD_EXECUTABLE):
  add_executable cannot create target "OpenApoc" because another target with
  the same name already exists.  The existing target is an executable created
  in source directory "/home/thee/games/OpenApoc/dependencies".  See
  documentation for policy CMP0002 for more details.


CMake Error at CMakeLists.txt:96 (target_link_libraries):
  Attempt to add link library "OpenApoc_Library" to target "OpenApoc" which
  is not built in this directory.


CMake Error at CMakeLists.txt:97 (target_link_libraries):
  Attempt to add link library "OpenApoc_Framework" to target "OpenApoc" which
  is not built in this directory.

```

**Comment by JonnyH** [comment]
"""
I have got an error for not having CMakeLists.txt in dependencies/physfs so I copied one frome the root directory
"""

This is wrong - it means it would try to build physfs using the wrong instructions, which could cause all kinds of weird errors. Likely including the ones you're seeing.

Are there any files in there? Have to updated the submodules? Does 'git status' say there are any changes? Either run from the root or the dependencies/physfs folder?

It's likely running "git submodule init" and "git submodule update" from the root of openapoc would fix this, as unless you've already run that (or cloned with --recursive) all the dependency directories would be empty, and it's just physfs was the first one that cmake tried to use and failed on.
